segmentstore/contracts/src/main/java/io/pravega/segmentstore/contracts/tables/TableEntry.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.contracts.tables;
+
+import io.pravega.common.util.ArrayView;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * An Entry in a Table Segment, made up of a Key and a Value, with optional Version.
```

Comment:
I have a question below about the version being optional.

Reply:
I answered it.